clients/richclient/src/main/java/org/openecard/richclient/RichClient.java

```diff
@@ -492,4 +503,61 @@ private static Long getRegInt(WinReg.HKEY hk, String key, String value, @Nullabl
 	return defaultValue;
     }
 
+    private static void setJnaRuntimeDiretory() throws IOException {
+	// read value of jna.tmpdir property
+	Properties properties = new Properties(System.getProperties());
+	String propJnaTmpDir = properties.getProperty("jna.tmpdir");
+
+	// if the property has been set externally don't change it
+	if (propJnaTmpDir != null) {
+	    return;
+	}
+
+	//check if we are on Linux
+	String osName = properties.getProperty("os.name");
+	if (osName.contains("nix") || osName.contains("nux") || osName.contains("aix")) {
+
+	    // we are on linux, first read "XDG_RUNTIME_DIR" to see if a user run dir is set
+	    String userRuntimeDir = System.getenv("XDG_RUNTIME_DIR");
+	    boolean canUseNormalTempDir = true;
+	    // if it is not set, we cannot use it
+	    boolean canUseUserRunDir = (userRuntimeDir != null);
+
+	    // then read "/proc/mounts"
+	    try (BufferedReader bufferedReader = new BufferedReader(new FileReader("/proc/mounts"))) {
+		String line;
+
+		while ((line = bufferedReader.readLine()) != null) {
+		    // split by whitespace to get the 6 parts individually
+		    String[] parts = line.split(" ");
+		    String mountPath = parts[1]; // get the path where the file system is mounted
+		    String mountOptions = parts[3]; // get the mount options
+
+		    if (mountPath.equals("/tmp")) {
+			// we can only use /tmp if the mountOptions do not contain "noexec"
+			canUseNormalTempDir = (!mountOptions.contains("noexec"));
+		    } else if (mountPath.equals(userRuntimeDir)) {
+			// same for the user run dir; if the user run dir is not set, then the equals above will be false
+			canUseUserRunDir = (!mountOptions.contains("noexec"));
+		    }
+		}
+	    }
+
+	    if (canUseNormalTempDir) {
+		// we can use /tmp directly and as JNA uses it as default anyway, nothing more to do here
+		return;
+	    }
+
+	    if (canUseUserRunDir) {
+		// the user run dir is set and executable, set jna.tempdir to XDG_RUNTIME_DIR
+		LOG.debug("setting jna.tmpdir to user run dir at {}", userRuntimeDir);
+		System.getProperties().putIfAbsent("jna.tmpdir", userRuntimeDir);
+	    } else {
+		// user run dir is not set or noexec as well, use '~/.openecard/run' as last ressort
+		LOG.debug("setting jna.tmpdir to be '~/.openecard/run' as last ressort");
+		System.getProperties().putIfAbsent("jna.tmpdir", "~/.openecard/run");
```

Comment:
Does this work without creating the directory?

Reply:
It seemed to work without. As I wasn't entirely sure why, I had a look into the JNA code and JNA indeed creates the directory referenced by "jna.tmpdir" itself, as long as this property is set (see https://github.com/java-native-access/jna/blob/3d9b024e672aa65e2a0f24438ce3196f00c1c2eb/src/com/sun/jna/Native.java#L1276).